[Bug] TestOverlayView  may fail with unexpected mount options

### Description

When running tests, it appears that the test `TestOverlayView` may result in the following error:

```
=== RUN   TestOverlayView
    snapshot_test.go:696: expected 1 mount option but got 2
--- FAIL: TestOverlayView (0.07s)
FAIL
```

### Steps to reproduce the bug

Unclear. It must be something specific to the test environment.

Please see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1103361 for the full conversation and context.

### Describe the results you expected

The test to pass

### Host information

1. OS:  Debian/trixie
2. Snapshotter Version:  0.4.1-4
3. Containerd Version: [1.7.24](https://packages.debian.org/source/testing/containerd)


### Any additional context or information about the bug

Applying the following patch:

```
-       if len(m.Options) != 1 {
-               t.Errorf("expected 1 mount option but got %d", len(m.Options))
-       }
+       if len(m.Options) != 1 {
+               t.Fatalf("expected 1 mount option but got %d: %+v", len(m.Options), m.Options)
+       }
```

shows this:

```
=== RUN   TestOverlayView
    snapshot_test.go:696: expected 1 mount option but got 2: [lowerdir=/tmp/TestOverlayView2389272231/001/snapshots/2/fs:/tmp/TestOverlayView2389272231/001/snapshots/1/fs userxattr]
```

I believe the first argument is expected, and the `userxattr` not?

--- a/snapshot/snapshot_test.go
+++ b/snapshot/snapshot_test.go
@@ -45,6 +45,7 @@ import (
 	"github.com/containerd/containerd/mount"
 	"github.com/containerd/containerd/pkg/testutil"
 	"github.com/containerd/containerd/snapshots"
+	"github.com/containerd/containerd/snapshots/overlay/overlayutils"
 	"github.com/containerd/containerd/snapshots/storage"
 	"github.com/containerd/containerd/snapshots/testsuite"
 	"github.com/containerd/errdefs"
@@ -709,12 +710,20 @@ func TestOverlayView(t *testing.T) {
 	if m.Source != "overlay" {
 		t.Errorf("mount source should be overlay but received %q", m.Source)
 	}
-	if len(m.Options) != 1 {
-		t.Errorf("expected 1 mount option but got %d", len(m.Options))
+	requiresUserXattrs, _ := overlayutils.NeedsUserXAttr("/tmp/view2")
+	expectedMountOptionsCount := 1
+	if requiresUserXattrs {
+		expectedMountOptionsCount = 2
+	}
+	if len(m.Options) != expectedMountOptionsCount {
+		t.Errorf("expected %d mount option(s) but got %d: %v", expectedMountOptionsCount, len(m.Options), m.Options)
 	}
 	lowers := getParents(ctx, o, root, "/tmp/view2")
 	expected = fmt.Sprintf("lowerdir=%s:%s", lowers[0], lowers[1])
 	if m.Options[0] != expected {
-		t.Errorf("expected option %q but received %q", expected, m.Options[0])
+		t.Errorf("expected option %q but got %q", expected, m.Options[0])
+	}
+	if requiresUserXattrs && m.Options[1] != "userxattr" {
+		t.Errorf("expected userxattr option, but got %s", m.Options[1])
 	}
 }

